server/node.go

```diff
@@ -390,6 +402,94 @@ func (n *Node) gossipCapacities() {
 	})
 }
 
+// storesScanner will walk through all the stores in the node every
+// ctx.ScanInterval and store the status in the db.
+func (n *Node) storesScanner(stopper *util.Stopper) {
+	stopper.RunWorker(func() {
+		start := time.Now()
+		for {
+			elapsed := time.Now().Sub(start)
+			remainingNanos := n.ctx.ScanInterval.Nanoseconds() - elapsed.Nanoseconds()
+			if remainingNanos < 0 {
+				remainingNanos = 0
+			}
+			nextIteration := time.Duration(remainingNanos)
+			log.V(6).Infof("next store scan iteration in %s", nextIteration)
+
+			select {
+			case <-time.After(nextIteration):
+				if !stopper.StartTask() {
+					continue
+				}
+
+				// Walk through all the stores on this node.
+				rangeCount := 0
+				stats := &proto.MVCCStats{}
+				storeIDs := n.lSender.GetStoreIDs()
+				accessedStoreIDs := make([]int32, 0, len(storeIDs))
+				for _, storeID := range storeIDs {
+					store, err := n.lSender.GetStore(storeID)
+					if err != nil {
+						// The rare case were a store was removed between
+						// listing the store ids and fetching their status.
+						log.Info("non-fatal error returned when scanning stores of node %v: %v", n.Descriptor.NodeID, err)
+						continue
+					}
+					storeStatus, err := store.GetStatus()
+					if err != nil {
+						log.Error(err)
+					}
+					if storeStatus == nil {
+						// The store scanner hasn't run on this node yet.
+						continue
```

Comment:
If you continue without setting 'start' again, it looks this will just run again immediately.